lib/flipper/adapters/moneta.rb

```diff
@@ -0,0 +1,160 @@
+require 'moneta'
+
+module Flipper
+  module Adapters
+    class Moneta
+      include ::Flipper::Adapter
+
+      FEATURES_KEY = :features
+
+      # Public: The name of the adapter.
+      attr_reader :name
+
+      # Public
+      def initialize(moneta)
+        @moneta = moneta
+        @name = :moneta
+      end
+
+      def features
+        read_feature_keys
+      end
+
+      # Public: Adds a feature to the set of known features.
+      def add(feature)
+        set_add(FEATURES_KEY, feature.key)
+        true
+      end
+
+      # Public: Removes a feature from the set of known features and clears
+      # all the values for the feature.
+      def remove(feature)
+        set_delete(FEATURES_KEY, feature.key)
```

Comment:
It would be neat if moneta supported transactions for data stores that did so we could do these two operations in a transaction, but I doubt it will be a problem in practice.